lib/splitclient-rb/sse/workers/splits_worker.rb

```diff
@@ -32,25 +32,44 @@ def stop
           SplitIoClient::Helpers::ThreadHelper.stop(:split_update_worker, @config)
         end
 
-        def add_to_queue(change_number)
-          @config.logger.debug("feature_flags_worker add to queue #{change_number}")
-          @queue.push(change_number)
+        def split_update(notification)
+          if @splits_repository.get_change_number() == notification.data['pcn']
```

Comment:
ideally you should check if you have correct data in `notification.data['d']`

Reply:
since it is inside an exception block, any issue with the data['d'] like encoding or compressing, will throw an exception.

Reply:
and do you have tests for this scenario ??

Reply:
Yep, in the integration PR